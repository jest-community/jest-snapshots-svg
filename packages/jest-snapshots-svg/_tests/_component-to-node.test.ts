import * as yoga from "yoga-layout"

import componentToNode, { styleFromComponent } from "../component-to-node"

describe("componentToNode", () => {
  it("generates the width for a simple component", () => {
    const component = {
      type: "View",
      props: {
        style: {
          width: 300,
          height: 40
        }
      },
      children: null
    }

    const settings = {
      width: 1024,
      height: 768
    }

    const node = componentToNode(component, settings)
    node.calculateLayout(yoga.UNDEFINED, yoga.UNDEFINED, yoga.DIRECTION_INHERIT)

    expect(node.getComputedWidth()).toEqual(300)
    expect(node.getComputedHeight()).toEqual(40)

    node.free()
  })
})

const componentWithStyle = (style: any) => ({ kind: "Stub", props: {style}})

describe("styleFromComponent", () => {
  it("handles deeply nested arrays of styles", () => {
    const style = [
      {
        fontSize: 30,
        color: "black",
        textAlign: "left",
        paddingLeft: 20,
        paddingRight: 20
      },
      [
        {
          textAlign: "center",
          fontSize: 30,
          lineHeight: 32,
          width: 280,
          marginTop: 35,
          alignSelf: "center"
        },
        null
      ],
      { fontFamily: "AGaramondPro-Regular" }
    ]
    const component = componentWithStyle(style)
    expect(styleFromComponent).toMatchSnapshot()
  })
})
