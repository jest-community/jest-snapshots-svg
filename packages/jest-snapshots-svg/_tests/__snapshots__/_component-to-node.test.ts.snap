// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styleFromComponent handles deeply nested arrays of styles 1`] = `[Function]`;
